config/eventhub/config.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventhub
+
+import (
+	"github.com/crossplane/terrajet/pkg/config"
+
+	"github.com/crossplane-contrib/provider-jet-azure/config/common"
+)
+
+// Configure configures resource group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_eventhub_namespace", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHubNamespace"
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Eventhub/namespaces/namespace1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+		)
+		r.LateInitializer = config.LateInitializer{
+			IgnoredFields: []string{"network_rulesets"},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_eventhub", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHub"
```

Comment:
nit: We had better not override the provider default configurator's behavior here. For being explicit, we may override it here (as the default configurator also uses the same kind name `EventHub` for this specific resource) but that would also mean that this resource's kind name will be handled as an exception from now on, making us unable to control behavior via the default configurator. For example, if for some reason we would like to rename all `.*Hub`s as `.*HUB`, then we will need extra care/maintenance for this resource. That's why I would prefer not overriding the default behavior unless needed.

Reply:
Got it, thanks for the explanation, it makes perfect sense